src/main/java/org/jenkinsci/plugins/workflow/support/pickles/ExecutorPickle.java

```diff
@@ -60,21 +67,29 @@ private ExecutorPickle(Executor executor) {
         }
     }
 
-    @Override public ListenableFuture<Executor> rehydrate() {
+    @Override public ListenableFuture<Executor> rehydrate(final FlowExecutionOwner owner) {
         return new TryRepeatedly<Executor>(1, 0) {
-            Future<Queue.Executable> future;
+            long itemID;
             @Override
             protected Executor tryResolve() throws Exception {
-                if (future == null) {
-                    Queue.Item item = Queue.getInstance().schedule2(task, 0).getItem();
+                Queue.Item item;
+                if (itemID == 0) {
+                    item = Queue.getInstance().schedule2(task, 0).getItem();
                     if (item == null) {
                         // TODO should also report when !ScheduleResult.created, since that is arguably an error
                         throw new IllegalStateException("queue refused " + task);
                     }
-                    future = item.getFuture().getStartCondition();
+                    itemID = item.getId();
+                } else {
+                    item = Queue.getInstance().getItem(itemID);
+                    if (item == null) {
+                        throw new IllegalStateException("queue lost item #" + itemID);
```

Comment:
`Queue.leftItems` keeps things for five minutes, much longer than the recurrence period for this pickle.